models/webhook.go

```diff
@@ -481,20 +567,60 @@ type HookEventType string
 
 // Types of hook events
 const (
-	HookEventCreate              HookEventType = "create"
-	HookEventDelete              HookEventType = "delete"
-	HookEventFork                HookEventType = "fork"
-	HookEventPush                HookEventType = "push"
-	HookEventIssues              HookEventType = "issues"
-	HookEventIssueComment        HookEventType = "issue_comment"
-	HookEventPullRequest         HookEventType = "pull_request"
-	HookEventRepository          HookEventType = "repository"
-	HookEventRelease             HookEventType = "release"
-	HookEventPullRequestApproved HookEventType = "pull_request_approved"
-	HookEventPullRequestRejected HookEventType = "pull_request_rejected"
-	HookEventPullRequestComment  HookEventType = "pull_request_comment"
+	HookEventCreate                    HookEventType = "create"
```

Comment:
Can't these constants be int?

Reply:
If I do that, then https://github.com/jolheiser/gitea/blob/granular_webhook/routers/api/v1/utils/hook.go#L106-L123 no longer works as-is, and I would need to make another function to translate the ints to explicit events.

Reply:
Or at least _some_ other function.  
It's possible, but it does create another place to look when modifying webhooks.